### PR TITLE
Improve the readability of the EventsByMonth and SegmentedControl components

### PR DESCRIPTION
**Pull request by alexwlchan**
This is hopefully the final bit of refactoring/clean-up in these components today. 🤞

### CODE REVIEW

File: common/views/components/SegmentedControl/SegmentedControl.tsx [hunk 1]
```diff
@@ -169,6 +185,35 @@ const SegmentedControl: FunctionComponent<Props> = ({
...
+    setActiveId && setActiveId(item.id);
+    isEnhanced && setIsActive(false);
+
+    // TODO: Do we still need this?  The original comment ("Assume we want to")
+    // didn't provide much context, and it's not clear if/how this changes the
+    // behaviour of the page, or indeed if it ever fires.
+    //
+    // The only page where we use SegmentedControl with anchors in the URL is
+    // the What's On page, and testing locally the `isHash` value will never
+    // be true – the `href` of the item is the full URL.
```
Comment (alexwlchan) [inline common/views/components/SegmentedControl/SegmentedControl.tsx:210]: I'm ~95% sure we could safely delete this, but I don't want to introduce yet more changes today and potentially break something else.

I did track down the commit which added it, but it doesn't have much more context: https://github.com/wellcomecollection/wellcomecollection.org/pull/3509/commits/71fc1867dde5bd6ae54c290c7767daa9b45eabbf

File: common/views/components/SegmentedControl/SegmentedControl.tsx [hunk 2]
```diff
@@ -169,6 +185,35 @@ const SegmentedControl: FunctionComponent<Props> = ({
...
+    const isHash = url.startsWith('#');
+
+    trackGaEvent({
+      category: 'SegmentedControl',
+      action: 'select segment',
+      label: item.text,
+    });
+
+    setActiveId && setActiveId(item.id);
+    isEnhanced && setIsActive(false);
```
Comment (alexwlchan) [inline common/views/components/SegmentedControl/SegmentedControl.tsx:202]: This is the only difference between the two branches; imo pulling them into a single function makes this clearer.

File: content/webapp/components/EventsByMonth/EventsByMonth.tsx [hunk 3]
```diff
@@ -49,29 +52,29 @@ const EventsByMonth: FunctionComponent<Props> = ({ events, links }) => {
...
+        .map(({ id, month, events }) => (
           <div
-            key={g.id}
+            key={id}
             className={cssGrid(gridSize12)}
             style={{ display: 'block' }}
           >
-            <h2 className="container" id={g.id}>
-              {g.month.month}
+            <h2 className="container is-hidden" id={id}>
```
Comment (alexwlchan) [inline content/webapp/components/EventsByMonth/EventsByMonth.tsx:70]: This got lost in a refactor -- previously it was being added with the condition `Boolean(activeId)`, which always evaluates to true, but I misread it as always evaluating to false.

The practical effect is that clicking on the tabs now causes the page to move around whenever you pick a month on the What's On page.

**Review by alexwlchan — COMMENTED** [review]
A few notes inline.

A sidebar: I did try retaining the per-month tabs when you reload the What's On page as part of this PR, e.g. if you click on June you're taken to `/whats-on#june-2023`, but reloading the page will drop you back on whatever the current default is. When I ran that, I got a warning that the client-side and server-side rendered pages were different.

(I'm also not convinced it's the right user experience – reloading you at the page drops you back at the top, and most users aren't looking at URL fragments. Why is it suddenly showing events for June would be more confusing than not imo.)

**Review by davidpmccormick — APPROVED** [review]

**Pull request merged**